integration-tests/physics/test_terminator_toy.py

```diff
@@ -0,0 +1,112 @@
+"""
+This tests the terminator toy physics scheme that models the interaction
+of two chemical species through coupled ODEs.
+"""
+
+from gusto import *
+from firedrake import IcosahedralSphereMesh, Constant, cos, \
+    sin, SpatialCoordinate, Function, max_value, as_vector, \
+    errornorm, norm
+import numpy as np
+
+
+def run_terminator_toy(dirname):
+
+    # ------------------------------------------------------------------------ #
+    # Set up model objects
+    # ------------------------------------------------------------------------ #
+
+    # A much larger timestep than in proper simulations, as this
+    # tests moving towards a steady state with no flow.
+    dt = 100000.
+
+    # Make the mesh and domain
+    R = 6371220.
+    mesh = IcosahedralSphereMesh(radius=R,
+                                 refinement_level=3, degree=2)
+
+    domain = Domain(mesh, dt, 'BDM', 1)
+
+    # get lat lon coordinates
+    x = SpatialCoordinate(mesh)
+    lamda, theta, _ = lonlatr_from_xyz(x[0], x[1], x[2])
+
+    domain = Domain(mesh, dt, 'BDM', 1)
+
+    # Define the interacting species
+    X = ActiveTracer(name='X', space='DG',
+                     variable_type=TracerVariableType.mixing_ratio,
+                     transport_eqn=TransportEquationType.advective)
+
+    X2 = ActiveTracer(name='X2', space='DG',
+                      variable_type=TracerVariableType.mixing_ratio,
+                      transport_eqn=TransportEquationType.advective)
+
+    tracers = [X, X2]
+
+    # Equation
+    V = domain.spaces("HDiv")
+    eqn = CoupledTransportEquation(domain, active_tracers=tracers, Vu=V)
+
+    output = OutputParameters(dirname=dirname+"/terminator_toy",
+                              dumpfreq=10)
+    io = IO(domain, output)
+
+    # Define the reaction rates:
+    theta_c = np.pi/9.
+    lamda_c = -np.pi/3.
+
+    k1 = max_value(0, sin(theta)*sin(theta_c) + cos(theta)*cos(theta_c)*cos(lamda-lamda_c))
+    k2 = 1
+
+    physics_schemes = [(TerminatorToy(eqn, k1=k1, k2=k2, species1_name='X',
+                        species2_name='X2'), BackwardEuler(domain))]
+
+    # Set up a non-divergent, time-varying, velocity field
+    def u_t(t):

```

Comment:
Since the wind is zero, do we need a function prescribing the velocity? Or could we just set the initial velocity to zero?

Reply:
I can't find a way to get the prescribed_transporting_velocity to work without giving it a function.